04-dom/exercise/09-numberConvertor.js

```diff
@@ -0,0 +1,59 @@
+function solve() {
+
+    let optionBinary = document.createElement("OPTION");
+    let optionHexadecimal = document.createElement("OPTION");
+
+    const input = document.getElementById('input');
+    const selectMenuTo = document.getElementById('selectMenuTo');
+    const btn = document.getElementsByTagName('button');
+    const result = document.getElementById('result');
+
+    optionBinary.textContent = 'Binary';
+    optionBinary.value = 'binary';
+    optionHexadecimal.textContent = 'Hexadecimal';
+    optionHexadecimal.value = 'hexadecimal';
+    selectMenuTo.appendChild(optionBinary);
+    selectMenuTo.appendChild(optionHexadecimal);
+
+    btn[0].onclick = function () {
+        if (selectMenuTo.value == 'binary') {
+            result.value = convertToBinary(Number(input.value));
+        } else if (selectMenuTo.value == 'hexadecimal') {
+            result.value = convertToHeXadecimal(Number(input.value));
+        }
+    }
+
+    function convertToBinary(num) {
+        let result = [];
+        while (num > 0) {
+            let digit = num % 2;
+            result.unshift(digit);
+            num = Math.floor(num / 2);
+        }
+        return result.join('');
+    }
+
+    function convertToHeXadecimal(num) {
```

Comment:
Of course this function works fine but if you have the wish you can research for some ways to improve it. I would suggest something like this (not tested, just a suggestion):

```javascript
function convertToHexadecimal(num) {
  // Base case: if num is 0, return '0' (special case)
  if (num === 0) return '0';

  // Define a string containing all hexadecimal characters in order
  const hexChars = '0123456789ABCDEF';

  // Initialize an empty string for the result
  let result = '';

  // Loop until num becomes 0
  while (num > 0) {
    // Calculate the remainder when dividing by 16 (hexadecimal base)
    const digit = num % 16;

    // Prepend the corresponding hexadecimal character to the result
    result = hexChars[digit] + result;

    // Reduce num by the integer division by 16
    num = Math.floor(num / 16);
  }

  // Return the hexadecimal representation of the original number
  return result;
}
```